src/math/trig.c

```diff
@@ -61,41 +140,53 @@ int32_t sin_fixed(int32_t th_rad_fxp)
 	 *without overflow.Increase of fractionLength makes the addition or
 	 *subtraction of a multiple of pi/2 more precise
 	 */
-	if (th_rad_fxp > CORDICSINE_PIOVERTWO_Q28FL) {
-		if ((th_rad_fxp - CORDICSINE_PIOVERTWO_Q29FL) <= CORDICSINE_PIOVERTWO_Q28FL) {
-			th_rad_fxp -= CORDICSINE_PIOVERTWO_Q29FL;
+	if (th_rad_fxp > cord_sincos_piovertwo_q28fl) {
```

Comment:
if this section is a duplicate above we should make an inner function and pass in the lookup table and relevant information

Reply:
Yes, optimized further including lookup table